include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,372 @@
+    //! Returns an aligned iterator to the beginning
+    EVE_FORCEINLINE auto begin_aligned() -> iterator_aligned { return views::convert(data_aligned(), eve::as<Type>{}); }
+
+    //! Returns an aligned iterator to the beginning
+    EVE_FORCEINLINE auto begin_aligned()  const -> const_iterator_aligned { return views::convert(data_aligned(), eve::as<Type>{}); }
+
+    //! Returns a constant aligned iterator to the beginning
+    EVE_FORCEINLINE auto cbegin_aligned() const -> const_iterator_aligned { return begin_aligned(); }
+
+    //! Returns an iterator to the beginning
+    EVE_FORCEINLINE auto begin() -> iterator { return views::convert(data(), eve::as<Type>{}); }
+
+    //! Returns an iterator to the beginning
+    EVE_FORCEINLINE auto begin()  const -> const_iterator { return views::convert(data(), eve::as<Type>{}); }
+
+    //! Returns a constant iterator to the beginning
+    EVE_FORCEINLINE auto cbegin() const -> const_iterator { return begin(); }
+
+    //! Returns an iterator to the end
+    EVE_FORCEINLINE auto end()  -> iterator  { return begin() + size(); }
+
+    //! Returns an iterator to the end
+    EVE_FORCEINLINE auto end()  const -> const_iterator { return begin() + size(); }
+
+    //! Returns a constant iterator to the end
+    EVE_FORCEINLINE auto cend() const -> const_iterator { return end(); }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Comparisons and ordering
+    //! @{
+    //==============================================================================================
+    friend bool operator==(soa_vector const& lhs, soa_vector const& rhs)
+    {
+      if( lhs.size() != rhs.size() ) return false;
+      return eve::algo::equal( lhs, rhs.begin_aligned());
+    }
+
+    friend bool operator!=(soa_vector const& lhs, soa_vector const& rhs)
+    {
+      return !(lhs == rhs);
+    }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    friend std::ostream& operator<<(std::ostream& os, soa_vector const& v)
+    {
+      os << "{ ";
+      for(std::size_t i=0;i<v.size();++i) os << v.get(i) << " ";
+      os << '}';
+      return os;
+    }
+
+    template <typename Traits, typename Self>
+    requires std::same_as<std::remove_reference_t<Self>, soa_vector>
+    EVE_FORCEINLINE friend auto tagged_dispatch(algo::preprocess_range_, Traits tr, Self& self)
+    {
+      return algo::preprocess_range(tr, algo::as_range(self.begin_aligned(), self.end()));
+    }
+
+    private:
+    soa_vector(storage_t&& s) : storage(std::move(s)) {}
```

Comment:
Why is this a thing???

Reply:
I use it internally to build a soa_vector from a storage to simplify the code in resize